File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/update_knowledge_base_source_update_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class UpdateKnowledgeBaseSourceUpdateDetails(object):
    """
    The information to be updated for KnowledgeBase source.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new UpdateKnowledgeBaseSourceUpdateDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param sources:
            The value to assign to the sources property of this UpdateKnowledgeBaseSourceUpdateDetails.
        :type sources: list[oci.aidataplatform_dp.models.UpdateKnowledgeBaseSourceUpdateDetailsItem]

        """
        self.swagger_types = {
            'sources': 'list[UpdateKnowledgeBaseSourceUpdateDetailsItem]'
        }

        self.attribute_map = {
            'sources': 'sources'
        }

        self._sources = None

    @property
    def sources(self):
        """
        Gets the sources of this UpdateKnowledgeBaseSourceUpdateDetails.
        Batch Payload for updating KnowledgeBase sources. Items marked for deletion are processed before items marked for addition


        :return: The sources of this UpdateKnowledgeBaseSourceUpdateDetails.
        :rtype: list[oci.aidataplatform_dp.models.UpdateKnowledgeBaseSourceUpdateDetailsItem]
        """
        return self._sources

    @sources.setter
    def sources(self, sources):
        """
        Sets the sources of this UpdateKnowledgeBaseSourceUpdateDetails.
        Batch Payload for updating KnowledgeBase sources. Items marked for deletion are processed before items marked for addition


        :param sources: The sources of this UpdateKnowledgeBaseSourceUpdateDetails.
        :type: list[oci.aidataplatform_dp.models.UpdateKnowledgeBaseSourceUpdateDetailsItem]
        """
        self._sources = sources

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/UpdateKnowledgeBaseSourceUpdateDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * The information to be updated for KnowledgeBase source.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=UpdateKnowledgeBaseSourceUpdateDetails.Builder.class)

public final class UpdateKnowledgeBaseSourceUpdateDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"sources"})
    public UpdateKnowledgeBaseSourceUpdateDetails(java.util.List<UpdateKnowledgeBaseSourceUpdateDetailsItem> sources) {
        super();
        this.sources = sources;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Batch Payload for updating KnowledgeBase sources. Items marked for deletion are processed before items marked for addition
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("sources")
        private java.util.List<UpdateKnowledgeBaseSourceUpdateDetailsItem> sources;

                /**
         * Batch Payload for updating KnowledgeBase sources. Items marked for deletion are processed before items marked for addition
         * @param sources the value to set
         * @return this builder
         **/
        

        public Builder sources(java.util.List<UpdateKnowledgeBaseSourceUpdateDetailsItem> sources) {
        this.sources = sources;
        return this;
        }


        public UpdateKnowledgeBaseSourceUpdateDetails build() {
            UpdateKnowledgeBaseSourceUpdateDetails model = new UpdateKnowledgeBaseSourceUpdateDetails(this.sources);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(UpdateKnowledgeBaseSourceUpdateDetails model) {
                this.sources(model.getSources());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Batch Payload for updating KnowledgeBase sources. Items marked for deletion are processed before items marked for addition
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("sources")
    private final java.util.List<UpdateKnowledgeBaseSourceUpdateDetailsItem> sources;

        /**
     * Batch Payload for updating KnowledgeBase sources. Items marked for deletion are processed before items marked for addition
     * @return the value
     **/
    
    public java.util.List<UpdateKnowledgeBaseSourceUpdateDetailsItem> getSources() {
        return sources;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("UpdateKnowledgeBaseSourceUpdateDetails(");
        sb.append("sources=").append(String.valueOf(this.sources));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof UpdateKnowledgeBaseSourceUpdateDetails)) {
            return false;
        }

        UpdateKnowledgeBaseSourceUpdateDetails other = (UpdateKnowledgeBaseSourceUpdateDetails) o;
        return java.util.Objects.equals(this.sources, other.sources);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.sources == null ? 43 : this.sources.hashCode());
        return result;
    }


}
